teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -145,59 +145,166 @@ See [TEP-0074](./0074-deprecate-pipelineresources.md) for the deprecation plan f
+* Functionality that could be supported with updated pod construction logic:
+  * [Step resource requirements](https://tekton.dev/docs/pipelines/tasks/#defining-steps)
+    * Any solution that runs multiple Tasks in one pod will need to determine how container resource requirements
+    should be set based on the resource requirements of each Task.
+  * [Sidecars](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-sidecars)
+    * We may need to wrap sidecar commands such that sidecars don't start executing until their corresponding Task starts
+    * We will also need to handle the case where multiple Tasks define sidecars with the same name
+* Functionality that would require additional orchestration within the pod (e.g. entrypoint changes):
+  * [Passing results between tasks](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#passing-one-tasks-results-into-the-parameters-or-whenexpressions-of-another)
+  * [retries](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-retries-parameter)
+  * Contextual variable replacement that assumes a PipelineRun, for example [`context.pipelineRun.name`](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-pipeline)
+  * [Parallel tasks](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#configuring-the-task-execution-order)
+  * [When expressions](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-whenexpressions)
+    (and [Conditions](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-conditions))
+  * [Finally tasks](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#adding-finally-to-the-pipeline)
+  * [Allowing step failure](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-onerror-for-a-step)
+* Functionality that would require significantly expanded orchestration logic:
+  * [Custom tasks](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-custom-tasks) - the pod would
+    need to be able to create and watch Custom Tasks, or somehow lean on the Pipelines controller to do this
+* Functionality that might not be possible (i.e. constrained by pods themselves):
+  * Any dynamically created TaskRuns. This is discussed in more detail [below](#dynamically-created-tasks-in-pipelines).
+  * Running each Task with a different `ServiceAccount` - the pod has one ServiceAccount as a whole
+
+(See also [functionality supported by experimental Pipeline to TaskRun](https://github.com/tektoncd/experimental/tree/main/pipeline-to-taskrun#supported-pipeline-features))
+
+### Dynamically created Tasks in Pipelines
+Currently, the number of Tasks in a Pipeline is determined at authoring time based on the number of items in a Pipeline's `Tasks` field.
+However, [TEP-0090: Matrix](./0090-matrix.md) proposes a feature that would allow additional Tasks to be created at runtime.
```

Comment:
nit: the number of tasks would stay the same, what changes is the number of taskruns

```suggestion
### Dynamically created TaskRuns in Pipelines
Currently, the number of TaskRuns created from a Pipeline is determined at authoring time based on the number of items in a Pipeline's `Tasks` field.
However, [TEP-0090: Matrix](./0090-matrix.md) proposes a feature that would allow additional TaskRuns to be created when a `Pipeline` is executed.
```